tracer/src/Datadog.Trace/Iast/Aspects/System/StringAspects.cs

```diff
@@ -987,11 +1444,20 @@ public static string[] Split(string target, string[] separator, StringSplitOptio
     public static string[] Split(string target, string[] separator, int count, StringSplitOptions options)
     {
         var result = target.Split(separator, count, options);
-        PropagationModuleImpl.PropagateResultWhenInputTainted(result, target);
+        try
+        {
+            PropagationModuleImpl.PropagateResultWhenInputTainted(result, target);
+        }
+        catch (Exception ex)
+        {
+            IastModule.Log.Error(ex, $"Error invoking {nameof(StringAspects)}.{nameof(Split)}");
+        }
+
         return result;
     }
 
-#if !NETFRAMEWORK
+#if NETCOREAPP2_1_OR_GREATER
```

Comment:
nit: we should probably stick to NETCOREAPP
```suggestion
#if NETCOREAPP
```

Reply:
The problem here is netcoreapp includes netcoreapp2.0, and many overloads are missing there and available from 2.1 (even we don't compile for that platform). This could be netcoreapp3_1_or_greater, but I don't loose the faith in changing  netstandard2 for netcoreapp2.1 someday

Reply:
> netcoreapp includes netcoreapp2.0

I see, so the concern is we'll add a .NET Core 2.0 target, then this would break. TBH. I don't think we care about that, because it's never going to happen 😄 But it's a nit anyway so all good